### ISSUE TITLE
Cross entropy loss

### ISSUE THREAD

**Issue by Naykira** [ildoonet/tf-pose-estimation#317]
Hello! Has anyone tried replacing the L2 loss with sigmoid cross entropy loss in the openpose implementation? I came across only one paper which uses the latter instead of former for human pose estimation. But surprisingly, it is giving better result with lower training time compared to L2 loss, at least for the heatmaps. PafMats / Vectormaps give very bad result though with Sigmoid cross entropy. Any idea why L2 loss is preferred over the cross entropy and also why it gives bad result for pafMats?

**Comment by ildoonet** [comment]
I'm not sure. Is it binary cross entropy?

**Comment by Naykira** [comment]
@luckhhn Yes, PafMats means Vectormaps.

**Comment by luckhhn** [comment]
@Naykira thank you!  But surprisingly, it is giving better result with lower training time compared to L2 loss, at least for the heatmaps. PafMats / Vectormaps give very bad result though with Sigmoid cross entropy. Any idea why L2 loss is preferred over the cross entropy and also why it gives bad result for pafMats

Is this the result of your run code?Have you solved your problem?Do you know how to do this if I want to map all the PafMats/Vectormaps?Looking forward to your reply

**Comment by Naykira** [comment]
@ildoonet Yes, it's binary cross entropy. The paper I refer to is 'Human pose estimation via Convolutional Part Heatmap Regression' by Bulat et al. (https://arxiv.org/pdf/1609.01743.pdf).

**Comment by Naykira** [comment]
@luckhhn Yes, that was the result of my code. What do you mean by 'Do you know how to do this if I want to map all the PafMats/Vectormaps?' Could you explain further?

**Comment by luckhhn** [comment]
@Naykira thank. I mean visualize all the PafMats/Vectormaps in one frame.

The dimension of PafMats/Vectormaps is [1,46,46,39], so I don't understand the meaning of 46*46 and 39. If 39 represents the vector diagram of 18 nodes, then 16*2=34 (I calculate it this way: 17 key points, so 16 edges, is 16*2=32), which makes me very confused, and I didn't find the data to confirm the meaning of these Numbers.So, I want to ask you, what does that mean, and how do you visualize that

**Comment by luckhhn** [comment]
@Naykira I have one more question, the better the training effect,  the heatmaps and PafMats/Vectormaps the same .Isn't it? 

Why do you come to this situation saying that using a binary cross loss function, a heatmap is good, and then pafmat is bad

**Comment by luckhhn** [comment]
@Naykira  Which network did you train from scratch?Cmu?, moblienet?Using the 2014coco dataset?What are your training parameters can you share?

**Comment by Naykira** [comment]
@luckhhn The image at [here](https://arvrjourney.com/human-pose-estimation-using-openpose-with-tensorflow-part-2-e78ab9104fc8) explain why the depth of Vectormaps is 39. 
 What I mean by heatmap is better with CE loss is that for my specific inputs the output for body joint locations was slightly more accurate when comapred to L2 loss. And I could not find any satisfying reasons as to why everyone uses only L2 loss for heatmaps in human pose estimation. In case of vectormaps, the results did not improve when CE loss was used. 
 Finally, I trained mobilenet from scratch with a custom dataset. So, I can't really help with training parameters.

**Comment by luckhhn** [comment]
@Naykira thanks , Do you label your own data set?and  If you have a heat map, why do you have vectormaps?

**Comment by Naykira** [comment]
@luckhhn Yes, I labelled the dataset. And well, Openpose uses both heatmaps and vectormaps. That's how/ why it's different from other pose estimation implementations.

**Comment by luckhhn** [comment]
@Naykira you are great！！In the openpose model, heatmaps and vectormaps are the most Shared items in the final human joint extraction task.Or which is easier to need?

**Comment by Naykira** [comment]
Pardon?

**Comment by luckhhn** [comment]
@Naykira sorry...
In the model of openpose, which heatmaps and vectormaps contribute the most to the final human joint extraction task?Or which is easier to need?

**Comment by Naykira** [comment]
I'd suggest giving the [original paper](https://arxiv.org/pdf/1611.08050.pdf) a read as that may help you better understand how the algorithm works.

**Comment by luckhhn** [comment]
@Naykira ok.Thank you for your kind reply!good night
